main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -464,24 +463,24 @@ object Inliner {
   }
 
   /**
-    * Performs beta-reduction on a local def `exp` applied to `exps`.
+    * Performs beta-reduction, binding `exps` as let-bindings.
     *
     * It is the responsibility of the caller to first visit `exps` and provide a substitution from the definition site
     * of `exp`.
     *
-    * [[betaReduceLocalDef]] creates a series of let-bindings
+    * [[betaReduce]] creates a series of let-bindings
     * {{{
     *   let sym1 = exp1;
     *   // ...
     *   let symn = expn;
-    *   exp'
+    *   exp
     * }}}
-    * where `symi` is the symbol of the i-th formal parameter and `exp'` is the body of the local def.
+    * where `symi` is the symbol of the i-th formal parameter and `exp` is the body of the function.
     *
     * Lastly, it visits the top-most let-binding, thus possibly removing the bindings.
     */
-  private def betaReduceLocalDef(exp: Expr.LocalDef, exps: List[Expr], loc: SourceLocation, ctx0: LocalContext)(implicit sym0: Symbol.DefnSym, sctx: SharedContext, root: OccurrenceAst.Root, flix: Flix): Expr = {
-    val bindings = exp.fparams.zip(exps).foldRight(exp.exp1) {
+  private def betaReduce(exp: Expr, exps: List[(FormalParam, Expr)], loc: SourceLocation, ctx0: LocalContext)(implicit sym0: Symbol.DefnSym, sctx: SharedContext, root: OccurrenceAst.Root, flix: Flix): Expr = {
```

Comment:
Just so we are clear: This is not beta reduction.

This is creating an expression that let binds the formal arguments.

Reply:
While it also visits the expression body.